### ISSUE TITLE
Should AllocateDescriptorSets take []DescriptorSet rather than *DescriptorSet?

### ISSUE THREAD

**Issue by Noofbiz** [vulkan-go/vulkan#35]
Like how CommandBuffers or other ones that have multiple names, they generally take a slice rather than a pointer. It's possible to do it this way in Vulkan in C++ as follows

```cpp
std::vector<VkDescriptorSet> descriptorSets;
vkAllocateDescriptorSets(device, &allocInfo, descriptorSets.data());
```

**Comment by xlab** [comment]
Yes, in most cases the difference between a reference and an array is not clear to the automated conveter. So it must be specified in the manifest — https://raw.githubusercontent.com/vulkan-go/vulkan/master/vulkan.yml

**Comment by celer** [comment]
+1
